spyder/config/manager.py

```diff
@@ -637,4 +641,21 @@ def reset_shortcuts(self):
             plugin_config.reset_to_defaults(section='shortcuts')
 
 
-CONF = ConfigurationManager()
+try:
+    CONF = ConfigurationManager()
+except Exception:
+    from spyder.app.utils import create_application
+    app = create_application()
+    reset_reply = QMessageBox.critical(
+        None, 'Spyder',
+        _("Error loading Spyder's preferences manager. "
+          "You will need to reset Spyder configuration files "
+          "for Spyder to be able to launch.\n\n"
+          "Reset the Spyder configuration files now?"),
+        QMessageBox.Yes, QMessageBox.No)

```

Comment:
Do you think it is important to allow users to select `No` here? I mean, is there any advantage in doing that?

Reply:
This will allow them to for example check by themselves the config files or do a manual backup. So, I think that with the `No` option the users have some more control over the situation instead to going directly to impose a forced reset/removal of their config files.